category.php

```diff
@@ -17,11 +17,24 @@
 
 $context = Timber::get_context();
 $taxonomy = get_queried_object();
+$featured_action = get_posts([
+    'post_type' => 'p4_action',
+    'category' => $taxonomy->term_id,
+    'orderby' => 'date',
+    'order' => 'DESC',
+    'numberposts' => 1,
+])[0] ?? null;
+$featured_action_id = $featured_action->ID ?? null;
+
 $context['taxonomy'] = $taxonomy;
 $context['wp_title'] = $taxonomy->name;
 $context['canonical_link'] = home_url($wp->request);
 $context['og_type'] = 'website';
 $context['og_description'] = $taxonomy->description;
+$context['featured_action'] = $featured_action;
+$context['featured_action_image'] = has_post_thumbnail($featured_action_id) ?
```

Comment:
Did you try setting the `$featured_action_id` to `null` and if the code fails??

Reply:
Hey @GP-Dan-Tovbein the tests were failing because we were assuming the `id` will be present and not taking into account if it was not, so I added this to fix that.